File: healthcare_app_server/chat_service/models/mariadb/friend.js
const friend_status = {
    PENDING: 0,
    ACCEPTED: 1,
    REJECTED: 2
}

class Friend {
    constructor(sender, receiver) {
        this._created_at = this._updated_at = new Date();
        this._sender = sender;
        this._receiver = receiver;
        this._status = friend_status.PENDING;
    }

    get sender() {
        return this._sender;
    }

    get receiver() {
        return this._receiver;
    }

    get status() {
        return this._status;
    }

    get created_at() {
        return this._created_at;
    }

    get updated_at() {
        return this._updated_at;
    }
}

File: healthcare_app_server/admin_service/src/main/java/dev/skyherobrine/service/models/mariadb/Friend.java
package dev.skyherobrine.service.models.mariadb;

import com.fasterxml.jackson.annotation.JsonFormat;
import dev.skyherobrine.service.enums.FriendStatus;
import dev.skyherobrine.service.keys.FriendKey;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity @Table(name = "friends")
@Getter @Setter
@NoArgsConstructor @RequiredArgsConstructor
public class Friend {

    @EmbeddedId @NonNull
    private FriendKey id;

    @Enumerated(EnumType.ORDINAL)
    @Column(nullable = false)
    private FriendStatus status;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "dd-MM-yyyy-HH-mm-ss")
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void onPersist() {
        createdAt = updatedAt = LocalDateTime.now();
        status = FriendStatus.PENDING;
    }

    @PreUpdate
    public void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
